docroot/modules/custom/examples/page_example/src/Controller/PageExampleController.php

```diff
@@ -13,6 +15,32 @@ class PageExampleController extends ControllerBase {
 
   use DescriptionTemplateTrait;
 
+  /**
+   * A logger instance.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $logger;
+
+  /**
+   * {@inheritDoc}
+   *
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger
+   *   A logger instance.
+   */
+  public function __construct(
+    LoggerChannelFactoryInterface $logger
+  ) {
+    $this->logger = $logger->get('page_example_module');
```

Comment:
This is not good example because logger is already part of parent class.

Reply:
fixed.